### PR TITLE
gh-102799: use sys.exception() instead of sys.exc_info() in tests

### PR DESCRIPTION
**Pull request by iritkatriel**
Now that 3.10 is no longer getting bugfixes, we can use sys.exception() without worrying about it impacting bugfix backports (it exists since 3.11).

This PR does the tests.

* Issue: gh-102799

### CODE REVIEW

File: Lib/test/test_asyncio/test_tasks.py
Comment (gvanrossum) [inline Lib/test/test_asyncio/test_tasks.py]: Why add the `isinstance()` call? The original test insists that you get _exactly_ `CancelledError`. I think that's useful to test for, since not all cancellation code is resilient to subclasses of `CancelledError`.

(In other cases below I don't mind.)

**Review by gvanrossum — APPROVED** [review]
One nit, no need to wait for another approval after you fix it.

File: Lib/test/test_asyncio/test_tasks.py
Comment (iritkatriel) [inline Lib/test/test_asyncio/test_tasks.py]:
```suggestion
                        and type(sys.exception()) is asyncio.CancelledError
```

**Pull request merged**
